src/status_im/contexts/wallet/wallet_connect/events/session_proposals.cljs

```diff
@@ -143,35 +143,52 @@
 (rf/reg-event-fx :wallet-connect/approve-session-success
  (fn [{:keys [db]} [session]]
    (log/info "Successfully approved WalletConnect session" session)
-   (let [total-connected-dapps (data-store/get-total-connected-dapps db)]
+   (let [total-connected-dapps (data-store/get-total-connected-dapps db)
+         dapp-name             (data-store/get-dapp-name session)]
      {:fx [[:dispatch [:wallet-connect/on-new-session session]]
            [:dispatch [:wallet-connect/reset-current-session-proposal]]
            [:dispatch [:wallet-connect/redirect-to-dapp (data-store/get-dapp-redirect-url session)]]
            [:dispatch
             [:centralized-metrics/track :metric/dapp-session-proposal
              {:action                :approved
-              :total_connected_dapps total-connected-dapps}]]]})))
+              :total_connected_dapps total-connected-dapps}]]
+           [:dispatch
+            [:toasts/upsert
+             {:type :positive
+              :text (i18n/label :t/wallet-connect-proposal-approved-toast {:dapp dapp-name})}]]]})))
 
 (rf/reg-event-fx :wallet-connect/approve-session-error
  (fn [_ [error]]
    (log/error "Wallet Connect session approval failed"
               {:error error
                :event :wallet-connect/approve-session})
-   {:fx [[:dispatch [:wallet-connect/reset-current-session-proposal]]]}))
+   {:fx [[:dispatch [:wallet-connect/reset-current-session-proposal]]
+         [:dispatch
+          [:toasts/upsert
+           {:type :negative
+            :text (i18n/label :t/wallet-connect-something-went-wrong)}]]]}))
 
 (rf/reg-event-fx
  :wallet-connect/reject-session-proposal
  (fn [{:keys [db]} [proposal]]
    (let [web3-wallet                      (get db :wallet-connect/web3-wallet)
          {:keys [request response-sent?]} (:wallet-connect/current-proposal db)
          networks                         (networks/get-proposal-networks (or proposal request))
-         rejected?                        (nil? proposal)]
+         rejected?                        (nil? proposal)
+         dapp-name                        (-> (data-store/get-session-dapp-metadata (or proposal
+                                                                                        request))
+                                              :name)]
      {:fx (concat
            (when-not response-sent?
              [[:effects.wallet-connect/reject-session-proposal
                {:web3-wallet web3-wallet
                 :proposal    (or proposal request)
-                :on-success  #(log/info "Wallet Connect session proposal rejected")
+                :on-success  (fn []
+                               (log/info "Wallet Connect session proposal rejected")
+                               (rf/dispatch [:toasts/upsert
+                                             {:type :positive
```

Comment:
Should this toast be of type `:positive` if the session proposal was rejected?

Reply:
Since this isn’t an error and the user intended to reject the session, we can confirm it was successfully rejected by `:positive` type IMO.